### PR TITLE
Issue #3: Amigos Page

### LINKED ISSUE #3: Create Amigos Page
Create the Amigos Page.

Ideas for this page:

Two tabs:

Amigos Tab: Lists all of the user's Amigos in alphabetical order.

Requests Tab: List all of the user's pending Amigo Requests for them to accept/decline.

There should also be a button to add a new Amigo.
For now, let's have this require you to select from existing amigos.
We may need to revisit to allow creating a new account here for a dependent, but for now they would need to separately create the user's account first.

### PR DESCRIPTION
**Pull request by mckoon**
Issue #3: Amigos Page

Implements Amigos Page.
Adds DoseAmigosUserService.
Adds DoseAmigosUser mocks.
Updates Mock Events to pull users from mock users.

Closes #3.

### CODE REVIEW

File: app/app.ts [hunk 1]
```diff
@@ -17,7 +18,8 @@ import {DoseTimeService} from "./dose-time-service/dose-time.service";
         providers: [
             AuthUserService,
             DoseEventService,
-            DoseTimeService
+            DoseTimeService,
+            DoseAmigosUserService
```
Comment (mckoon) [inline app/app.ts]: Adds a provider to the app for the new `DoseAmigosUserService`.

File: app/dose-amigos-user-service/dose-amigos-user-mocks.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        lastTimeDoseTaken: fifteenMinAgo,
+        nextTimeDoseScheduled: now
+    },
+    {
+        id: 100,
+        name: "Betty Rubble",
+        lastTimeDoseTaken: fortyFiveMinAgo,
+        nextTimeDoseScheduled: fifteenMinAgo
+    }
+];
```
Comment (mckoon) [inline app/dose-amigos-user-service/dose-amigos-user-mocks.ts]: Mock amigos.

File: app/dose-amigos-user-service/dose-amigos-user.service.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    /**
+     * Get a list of DoseAmigosUsers.
+     * @returns {Promise<Array<DoseAmigosUser>>}
+     */
+    public list(): Promise<Array<DoseAmigosUser>> {
+        return Promise.resolve(
+            DOSE_AMIGOS_USERS
+        );
+    }
```
Comment (mckoon) [inline app/dose-amigos-user-service/dose-amigos-user.service.ts]: Method to list a user's amigos. Returns a promise-wrapped array of mocks now.

File: app/dose-event-service/dose-event.mocks.ts [hunk 4]
```diff
@@ -1,39 +1,20 @@
...
         id: 3,
-        time: now,
-        user: {
-            id: 300,
-            name: "Barney Rubble",
-            lastTimeDoseTaken: now,
-            nextTimeDoseScheduled: fifteenMinFromNow
-        }
+        time: DOSE_AMIGOS_USERS[0].lastTimeDoseTaken,
+        user: DOSE_AMIGOS_USERS[0]
```
Comment (mckoon) [inline app/dose-event-service/dose-event.mocks.ts]: Updated this mock DoseEvents object to use the mock DoseAmigosUsers for its users.

File: app/pages/amigos/amigos.html [hunk 5]
```diff
@@ -1,5 +1,9 @@
 <dose-amigos-toolbar [title]="title"></dose-amigos-toolbar>
 
 <ion-content padding class="amigos">
-  <div>Hola Amigos!</div>
+
+    <user-status-card *ngFor="let user of doseAmigosUsers"
+                      [user]="user"
+                      [onClickPage]="userStatusClickPage"></user-status-card>
```
Comment (mckoon) [inline app/pages/amigos/amigos.html]: Uses the same user-status-card we show at the top of the News Feed page for each of the current user's Amigos.

This allows a quick at-a-glance view of if each of your Amigos is on track:

amigos

File: app/pages/amigos/amigos.ts [hunk 6]
```diff
@@ -1,19 +1,44 @@
...
+
+    }
+
+    /**
+     * Initialize the page.
+     * @returns {Promise}
+     */
+    ngOnInit(): any {
+
+        this.userStatusClickPage = FeedPage;
```
Comment (mckoon) [inline app/pages/amigos/amigos.ts]: For now, clicking on an Amigo takes you to the News Feed Page.

We will want this to eventually take you to a Amigo Feed Page (news feed filtered to only a specific amigo's events).

**Comment by jking31cs** [comment]
Everything looks fine to me. 👍

**Comment by mckoon** [comment]
@geoffreygraddy verbally approved.
Merging.

**Pull request merged**